lib/OnyxCache.js

```diff
@@ -144,6 +154,34 @@ class OnyxCache {
 
         return this.pendingPromises[taskName];
     }
+
+    /**
+     * @private
+     * Adds a key to the top of the recently accessed keys
+     * @param {string} key
+     */
+    addToAccessedKeys(key) {
+        // Removing and re-adding a key ensures it's at the end of the list
+        this.recentKeys.delete(key);
+        this.recentKeys.add(key);
+    }
+
+    /**
+     * Remove keys that don't fall into the range of recently used keys
+     * @param {number} recentKeysSize - a list of most recent keys from this size will remain in cache.
+     */
+    removeLeastRecentUsedKeys(recentKeysSize) {
+        // Get the last N keys by doing a negative slice
+        const recentlyAccessed = new Set([...this.recentKeys].slice(-recentKeysSize));
+
+        _.chain(this.storageMap)
+            .keys()
+            .each((key) => {
+                if (!recentlyAccessed.has(key)) {
+                    this.drop(key);
+                }
+            });
```

Comment:
Rather than loop over every key it might be clearer to get the [difference](https://underscorejs.org/#difference) of these keys first then just drop all of them?

Reply:
Good idea, thanks